cobra/flux_analysis/loopless.py

```diff
@@ -1,6 +1,71 @@
 from ..core import Reaction, Metabolite
 from ..manipulation.modify import convert_to_irreversible
+import numpy as np
 from six import iteritems
+from cobra.util import nullspace, add_to_solver
+from sympy.core.singleton import S
+
+
+def add_loopless(model, zero_cutoff=1e-12):
+    """Add variables and constraints to make a model thermodynamically
+    feasible. This removes flux loops. The used formulation is described
+    in [1]_.
+
+    Parameters
+    ----------
+    model : a cobra model
+        The model to which to add the constraints.
+    zero_cutoff : positive float, optional
+        Cutoff used for null space. Coefficients with an absolute value smaller
+        than `zero_cutoff` are considered to be zero.
+
+    Returns
+    -------
+    Nothing
+
+    References
+    ----------
+    .. [1] Elimination of thermodynamically infeasible loops in steady-state
+       metabolic models.
+       Schellenberger J, Lewis NE, Palsson BØ.
+       Biophys J. 2011 Feb 2;100(3):544-53.
+       doi: 10.1016/j.bpj.2010.12.3707.
+       Erratum in: Biophys J. 2011 Mar 2;100(5):1381.
+    """
+    internal = [i for i, r in enumerate(model.reactions) if not r.boundary]
+    Sint = model.S[:, np.array(internal)]
```

Comment:
Do we think this is an operation we'll do many times? Just thinking that we could speed things up if I had the array-generating function accept an optional argument to only give you certain columns...

In this case though, probably not a speed-limiting factor.